csrc/device_lower/pass/index.cpp

```diff
@@ -1742,10 +1743,240 @@ static MmaInputSmemSwizzle getSwizzleMode(TensorView* tv) {
+// That is, if we futher split k_inst as:
+//      M            K
+//      |            |
+//     ...          ...
+//      |            |
+//     / \.         / \.
+//   ...  m_inst  ...  k_inst
+//                     /    \.
+//               linear      swizzle_size
+// Then we would need to prove that `linear` is linear in the allocation domain
+// of tv, and the stride of `linear` is the leading_bytes. This function does
+// the following things:
+// 1. Find k_inst.
+// 2. Split k_inst as above.
+// 3. Prove that `linear` is linear in the allocation domain of tv, and get the
+//    stride of `linear`.
+Val* getInnerStride(TensorView* tv, const MmaOp* mma) {
+  auto swizzle = getSwizzleMode(tv);
+  auto swizzle_size = getBytesFromSwizzle(swizzle) / dataTypeSize(tv->dtype());
+  ValGraph& id_graph = GpuLower::current()->tensorIndexer().traversalGraph();
+  auto alloc_domain = id_graph.toGroups(tv->getMaybeAllocationDomain());
+  auto inner = getInnerMmaLoopGroup(tv, mma);
+  // At this point, we can just create the following schedule:
+  //        inner
+  //       /     \.
+  //   linear   swizzle_size
+  // and use proveLinearAndGetStride to find the stride of `linear` in the
+  // allocation domain.
+  auto outer_of_tiling = split(&id_graph, inner, swizzle_size).first;
+  auto stride = lower_utils::proveLinearAndGetStride(
+      id_graph, outer_of_tiling, alloc_domain);
+  NVF_ERROR(stride != nullptr, "Could not get the stride of tiling");
+  return SimplifyingIrBuilder::mulExpr(stride, dataTypeSize(tv->dtype()));
+}
+
+// Compute the "stride_bytes" in the matrix descriptor of Mma. The stride
+// bytes is the stride of the outer dimension in the allocation domain of tv
+// considering core matrices. For example, if the tv is [M, K], where K is the
+// inner, then the schedule of the loop domain of the mma output must be
+// something like:
+//      M            K
+//      |            |
+//     ...          ...
+//      |            |
+//     / \.         / \.
+//   ...  m_inst  ...  k_inst
+// where m_inst and k_inst are the instruction tiles of M and K, respectively,
+// that is, the number of items each TensorCore instruction can execute. Both
+// m_inst and k_inst must be parallelized on Mma. The stride_bytes is the
+// stride of the outer (m_inst/8) in the allocation domain of tv.
+// That is, if we futher split m_inst as:
+//       M            K
+//       |            |
+//      ...          ...
+//       |            |
+//      / \.         / \.
+//    ...  m_inst  ...  k_inst
+//         /    \.
+//   linear      8
+// Then we would need to prove that `linear` is linear in the allocation domain
+// of tv, and the stride of `linear` is the stride_bytes. This function does
+// the following things:
+// 1. Find m_inst.
+// 2. Split m_inst as above.
+// 3. Prove that `linear` is linear in the allocation domain of tv, and get the
+//    stride of `linear`.
+Val* getOuterStride(TensorView* tv, const MmaOp* mma) {
```

Comment:
Should we name this `getOuterStrideBytes`?

Reply:
Renamed